doc: Add The Things Uno quickstart (#496)
TheThingsIndustries/lorawan-stack-docs: doc/content/devices/the-things-uno/_index.md

---
title: "The Things Uno"
description: ""
weight: 
---

> {{< figure src="TheThingsUno.png" alt="CloudGate Gateway" class="float plain" >}}

Reviewer: ```suggestion
{{< figure src="TheThingsUno.png" alt="The Things Uno" class="float plain" >}}
```

Author: fixed

**The Things Uno** lets you easily get started with **LoRaWAN** application development using the Arduino IDE. Choose the correct Things Uno board to suit the recommended frequency plan of your country/region.

## Setting up Arduino IDE

The Arduino Integrated Development Environment (IDE) allows you to write code using the Arduino programming language (C/C++), compile (verify), and upload it to the Things Uno board. It supports Windows, Mac OS, Linux, and Chrome OS.

Download the latest release of the [Arduino Software (IDE)](https://www.arduino.cc/en/Main/Software) and [install](https://www.arduino.cc/en/Guide) it on your operating system.

Start the Arduino Software (IDE).

In the Arduino IDE, select **Sketch &#8594; Include Library &#8594; Manage Libraries...** or **Tools &#8594; Manage Libraries...** from the menu bar.

In the **Library Manager** window, search for **TheThingsNetwork**.

Select **TheThingsNetwork** Arduino library from the search results.

Select the **Install** button.

When the installation finishes, close the **Library Manager** by selecting the **Close** button.

{{< figure src="LibraryManager.png" alt="Library Manager" >}}

> ## Connecting Things UNO to your Computer

Reviewer: ```suggestion
## Connecting Things Uno to your Computer
```

Fix this everywhere

    }

  debugSerial.println("-- STATUS");
  ttn.showStatus();

}
 
void loop()
{
}
```

Make sure that your Things Uno board is connected to your computer and select the **Upload** button in the toolbar. The IDE verifies your sketch again and uploads it to your Things Uno. During this process, the **TX/RX** LEDs on your Things Uno board should blink, indicating the information is traveling between the Things Uno and your computer.

Once uploaded, your sketch will immediately start to run but you cannot see any output unless you open the Arduino **Serial Monitor**. The Arduino Serial Monitor is a tool that can be used to print information passing through the serial port. 

In the Arduino IDE, select **Tools &#8594; Serial Monitor** from the menu bar. Once open the Serial Monitor prints something similar to the following output.

```
-- STATUS
EUI: 0004A30B001B7AD2
Battery: 3223
AppEUI: 70B3D57EF000001C
DevEUI: 0004A30B001B7AD2
Band: 868
Data Rate: 5
RX Delay 1: 1000
RX Delay 2: 2000
Total airtime: 0.00 s
```

**Copy** the Serial Monitor output to a text editor (E.g. Notepad) because you will need the **AppEUI** and **DevEUI** when you register your Things Node with {{% tts %}}.

> ## Registering The Things UNO with {{% tts %}}

Reviewer: ```suggestion
## Registering The Things Uno with {{% tts %}}
```

Author: done

- AppKey – Select **Generate** button to generate an **AppKey**.
- End device ID – Give your device a unique human-readable [identifier]({{< ref "reference/id-eui-constraints" >}}).

Select the **Register end device** button. 

{{< figure src="RegisterEndDeviceRepoOTAA.png" alt="Register End Device from LoRaWAN repository OTAA" >}}

If you have selected `abp` from the **Firmware Ver.** in the previous step:

- Frequency plan – Select `Europe 863-870 MHz (SF9 for RX2 - recommended)` for **EU_863_870** or `United States 902-928 MHz, FSB2 (used by TTN)` for **US_902_928**.
- Device address - Select **Generate** button.
- AppSKey– Select **Generate** button.
- NwkSKey– Select **Generate** button.
- End device ID – Give your device a unique human-readable [identifier]({{< ref "reference/id-eui-constraints" >}}).

Select the **Register end device** button. 

{{< figure src="RegisterEndDeviceRepoABP.png" alt="Register End Device from LoRaWAN repository ABP" >}}

Once registered, you will be redirected to the overview page of the newly registered device, where you can find the generated **AppKey** which we’ll need next.

> ### Manually

Reviewer: ```suggestion
### Manually
```

Let's remove the manual section. There are other instructions for manually adding devices, this isn't necessary.

Reviewer: @pradeeka7 please remove this whole "Manually" section.

Author: fixed

If you have selected `Over the air activation (OTAA)`:

- DevEUI - copy **DevEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- AppEUI - copy **AppEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- AppKey - select **Generate** button.
- End device ID – Give your device a unique human-readable [identifier]({{< ref "reference/id-eui-constraints" >}}).

{{< figure src="RegisterEndDeviceManualOTAA.png" alt="Register End Device Manually OTAA" >}}

If you have selected `Activation by personalization (ABP)`:

- DevEUI - copy **DevEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- Device address - select **Generate** button.
- AppSKey - select **Generate** button.
- NwkSKey - select **Generate** button.
- End device ID - Give your device a unique human-readable [identifier]({{< ref "reference/id-eui-constraints" >}}).

{{< figure src="RegisterEndDeviceManualABP.png" alt="Register End Device Manually ABP" >}}

Select the **Register end device** button.

Your device is now registered with {{% tts %}}! You will be redirected to the overview page of the Things Uno.

> ## Activating the Things UNO - OTAA

Reviewer: ```suggestion
## Activating the Things Uno - OTAA
```

---
title: "The Things Uno"
description: ""
weight: 
---

{{< figure src="TheThingsUno.png" alt="CloudGate Gateway" class="float plain" >}}

> The [Things Uno](https://www.thethingsnetwork.org/docs/devices/uno/) is based on the [Arduino Leonardo](https://store.arduino.cc/usa/leonardo) (not the Arduino Uno) with an added **Microchip** LoRaWAN module ([RN2483](https://www.microchip.com/wwwproducts/en/RN2483) or [RN2903](https://www.microchip.com/wwwproducts/en/RN2903)). It is fully compatible with the **Arduino IDE** and existing Arduino shields.

Author: done

**The Things Uno** lets you easily get started with **LoRaWAN** application development using the Arduino IDE. Choose the correct Things Uno board to suit the recommended frequency plan of your country/region.

## Setting up Arduino IDE

> The Arduino Integrated Development Environment (IDE) allows you to write code using the Arduino programming language (C/C++), compile (verify), and upload it to the Things Uno board. It supports Windows, Mac OS, Linux, and Chrome OS.

Author: done

Download the latest release of the [Arduino IDE](https://www.arduino.cc/en/Main/Software). There are different ways to install it. The full instructions can be found at https://www.arduino.cc/en/Guide for each operating system. After you have downloaded and installed the Arduino IDE, you will need to start the IDE.

In the **menu bar** select **Sketch &#8594; Include Library &#8594; Manage Libraries...** or **Tools &#8594; Manage Libraries...**

In the **Library Manager** window search for **The Things Network**.

Select **TheThingsNetwork** Arduino library from the search results.

Select the **Install** button.

When the installation finishes, close the **Library Manager** by selecting the **Close** button.

{{< figure src="LibraryManager.png" alt="Library Manager" >}}

## Connecting Things UNO to your Computer

> To begin, connect your Things Uno to the computer with the USB cable. 

Reviewer: IDEM

> [INSERT IMAGE]

Reviewer: These images you can just take from the TTN doc, they are not TTS specific

Author: no image was found for connecting the thins uno with a computer.

In the **menu bar** select **Tools &#8594; Board &#8594; Arduino Leonardo**. This will tell your Arduino IDE to upload sketches to your Things Uno board.

[INSERT IMAGE]

In the **menu bar** select **Tools &#8594; Port &#8594; Serial port** (the serial port associated with your Things Uno).

[INSERT IMAGE]

## Creating an Application

> The Things Uno (or any end device) needs to be registered with an **application** to communicate with The Things Stack. To create an application follow the steps given below:

Reviewer: IDEM

Author: done

> If you don’t have a Things Stack account, create a [free account](https://account.thethingsnetwork.org/register) before you begin. Next, sign in to [The Things Stack](https://eu1.cloud.thethings.network) with your credentials.

Reviewer: ```suggestion
If you don’t have a Things Stack account, create a [free account](https://account.thethingsnetwork.org/register) before you begin. Next, sign in to [The Things Stack](https://eu1.cloud.thethings.network) with your credentials.
```

Remove this section - don't need to direct people to create an account here

Author: removed

> In the top menu, select **Applications**, or in the middle of the window, select the **Go to applications** (large) icon.

Reviewer: Remove the instructions for adding an application. You can just link to integrations/adding-applications/ (use the ref shortcode)

Author: removed

{{< figure src="TheThingsStackConsole1.png" alt="The Things Stack Console" >}}

> On the **Applications** page, create a new application by selecting **+Add application** at the top-right of The Things Stack window.

Reviewer: Use the {{% tts %}} shortcode for The Things Stack (everywhere)

Author: done

In this section, you will learn how to write and upload your first sketch to The Things Uno. This sketch prints **DevEUI**, **AppEUI**, and some useful information that you will need to register your Things Uno with the Things Stack.

In the menu bar, select **File &#8594; New**. A new window will open with the default Arduino sketch.

```
void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}
```
In the menu bar select **Sketch &#8594; Include Library &#8594; The Things Network**. The following code will add to the top of the sketch.

```
#include <TheThingsNetwork.h>
```

This will import **The Things Network Arduino library** to your sketch and provide you with some useful functions to work with LoRaWAN. 

Type the following lines to create two serial port objects for **Serial** and **Serial1**.

```
#define loraSerial Serial1
#define debugSerial Serial
```
 
{{< note >}}The **Serial** object allows communication between **The Things Uno** and your **computer**. Also, the **Serial1** object allows communication between **The Things Uno** and the **Microchip LoRa module**.{{</ note >}}

> Type the following line to define the frequency plan of your Things Uno. Replace `REPLACE_ME` with `TTN_FP_EU868` or `TTN_FP_US915` depending on the frequency plan of your Things Uno and your country/region. (Click [here](https://www.thethingsnetwork.org/docs/devices/arduino/usage/#set-the-frequency-plan) to find the macro values for `freqPlan`).

Reviewer: Remove all links to TTN docs.

```
-- STATUS
EUI: 0004A30B001B7AD2
Battery: 3223
AppEUI: 70B3D57EF000001C
DevEUI: 0004A30B001B7AD2
Band: 868
Data Rate: 5
RX Delay 1: 1000
RX Delay 2: 2000
Total airtime: 0.00 s
```

**Copy** the Serial Monitor output to a text editor (i.e. Notepad) because you will need the **AppEUI** and **DevEUI** when you register your Things Node with the Things Stack.

## Registering The Things UNO with The Things Stack
It’s time to register your Things Uno with the Things Stack. 

On the **Applications** page, select your application to go to its overview page. 

Select **+ Add end device** in the bottom-right of the page.

{{< figure src="AddEndDevice.png" alt="Add end device" >}}

The Things Stack provides two options to register your end device.
- From the LoRaWAN Device Repository
- Manually

Each registering option provides both **OTAA** and **ABP** activation methods. Let’s have a look at them one by one.

### From the LoRaWAN Device Repository 
This is the easiest way to register your Things Uno with the Things Stack.

On the **Register end device** page, in the **From the LoRaWAN Device Repository** tab, under **Select the end device**, select the following mandatory fields:

> - Brand - `The Things Products`

Reviewer: Newline between paragraph and bullets

Author: done

- Model – `The Things Uno`
- Hardware Ver. – `1.0`
- Firmware Ver. – `quickstart` or `abp` (we recommend you to use `quickstart` as it provides **OTAA**.)
- Profile (Region) – Choose `EU_863_870` or `US_902_928` to match with your board. 

{{< figure src="RegisterEndDeviceRepo.png" alt="Register End Device from LoRaWAN repository" >}}

Under the **Enter registration data**, select/fill the following mandatory fields.

If you have selected `quickstart` from the **Firmware Ver.** in the previous step:
- Frequency plan – Select `Europe 863-870 MHz (SF9 for RX2 - recommended)` for **EU_863_870** or `United States 902-928 MHz, FSB2 (used by TTN)` for **US_902_928**.
- AppEUI – Copy the **AppEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- DevEUI – Copy the **DevEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- AppKey – Select **Generate** button to generate an **AppKey**.

> - End device ID – A unique human-readable identifier for your Things Uno.

Reviewer: ```suggestion
- End device ID – Give your device a unique human-readable identifier.
```

Author: fixed

> {{< note >}}

Reviewer: Don't copy and paste this here. Use a link, i.e {{< ref "reference/id-eui-constraints" >}}

Author: done

{{< figure src="RegisterEndDeviceRepoOTAA.png" alt="Register End Device from LoRaWAN repository OTAA" >}}

If you have selected `abp` from the **Firmware Ver.** in the previous step:
- Frequency plan – Select `Europe 863-870 MHz (SF9 for RX2 - recommended)` for **EU_863_870** or `United States 902-928 MHz, FSB2 (used by TTN)` for **US_902_928**.
- Device address - Select **Generate** button.
- AppSKey– Select **Generate** button.
- NwkSKey– Select **Generate** button.
- End device ID – A unique human-readable identifier for your Things Uno (see above for naming conversions).

Select the **Register end device** button. 

{{< figure src="RegisterEndDeviceRepoABP.png" alt="Register End Device from LoRaWAN repository ABP" >}}

Once registered, you will be redirected to the overview page of the newly registered device, where you can find the generated **AppKey** which we’ll need next.

### Manually
Manual registration requires providing more information than the previous option but it provides a flexible way of adding end devices. 

On the **Register end device** page select the **Manually** tab.

In the **Basic settings** select/fill the following:

> - LoRaWAN version - `MAC V1.1`

Reviewer: ```suggestion
- LoRaWAN version - `MAC V1.1`
```

This is incorrect, Uno uses MAC V1.0.2 and PHY 1.0.2 Rev B

Author: corrected

- Regional Parameters version: `PHY V1.1 REVB`
- Frequency plan – select the frequency plan for your region. For Europe select `Europe 863-870 MHz (SF9 for RX2 – recommended)` and for the US select `United States 902-923 MHz, FB2 (used by TTN)`.
- Select **Show advanced activation, LoRaWAN class and cluster settings** to expand the hidden section if not visible.
- Activation mode: `Over the air activation (OTAA)` or `Activation by personalization (ABP)`(we recommend you to use `Over the air activation (OTAA)`).

If you have selected `Over the air activation (OTAA)`:
- Additional LoRaWAN class capabilities - `None (class A only)`
- DevEUI - copy **DevEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- JoinEUI - copy **AppEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- AppKey - select **Generate** button.

> - NwkKey - select **Generate** button.

Reviewer: ```suggestion
- NwkKey - select **Generate** button.
```

This key doesnt exist in LoRaWAN 1.0.2

Author: corrected

- End device ID – A unique, human-readable identifier for your Things Uno (see above for naming conversions).

{{< figure src="RegisterEndDeviceManualOTAA.png" alt="Register End Device Manually OTAA" >}}

If you have selected `Activation by personalization (ABP)`:
- DevEUI - copy **DevEUI** from the output printed by the ***TheThingsUnoTest*** sketch.
- Device address - select **Generate** button.
- AppSKey - select **Generate** button.

> - FNwkSIntKey - select **Generate** button.

Reviewer: ```suggestion
- FNwkSIntKey - select **Generate** button.
```

These keys dont exist in 1.0.2

Author: fixed
